src/main/java/com/sleekbyte/tailor/format/JSONFormatter.java

```diff
@@ -79,9 +79,10 @@ public void displaySummary(long numFiles, long numSkipped, long numErrors, long
         System.out.println(GSON.toJson(output));
     }
 
-    private void storeMessages(List<Map<String, Object>> violations, boolean parsed) throws IOException {
+    private void storeMessages(List<Map<String, Object>> violations, boolean parsed, String filePath)
+        throws IOException {
         Map<String, Object> output = new HashMap<>();
-        output.put(Messages.PATH_KEY, inputFile.getCanonicalPath());
+        output.put(Messages.PATH_KEY, filePath);
         output.put(Messages.VIOLATIONS_KEY, violations);
         output.put(Messages.PARSED_KEY, parsed);
         FILES.add(output);
```

Comment:
this isn't thread safe, probably need something different from `ArrayList`, or alternatively, make this method `synchronized`